### PR TITLE
Fix #2141: Decouple OpenShift Model dependency from Kubernetes Client

### LINKED ISSUE #2141: Decouple openshift-model dependency from kubernetes-client
## Description
After model split (#2108 #2137) we have to keep a dependency to `openshift-model` in `kubernetes-client` as some of the classes still import OpenShift specific model types (`io.fabric8.openshift.api.model.*` e.g. `ClusterOperationsImpl`)

Classes with such imports should segregated with resulting classes moved to `openshift-client` module.

## Blocked by
- #2137

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fix #2141: Decouple OpenShift Model from Kubernetes Client

This change may be breaking for users who might be using KubernetesClient for OpenShift operations(like we were doing in our tests)

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ClusterOperationsImpl.java [hunk 1]
```diff
@@ -17,29 +17,24 @@
...
 import java.util.HashMap;
 import java.util.Map;
 
 public class ClusterOperationsImpl extends OperationSupport {
-  private String versionEndpoint;
+  protected final String versionEndpoint;
   public static final String KUBERNETES_VERSION_ENDPOINT = "version";
-  public static final String OPENSHIFT_VERSION_ENDPOINT = "version/openshift";
-  public static final String OPENSHIFT4_VERSION_ENDPOINT = "apis/config.openshift.io/v1/clusterversions";
   public static final ObjectMapper objectMapper = new ObjectMapper();
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ClusterOperationsImpl.java:38]: We should remove this mapper and either user `Serialization.jsonMapper` or any of the appropriate helper methods in that class.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ClusterOperationsImpl.java [hunk 2]
```diff
@@ -50,48 +45,34 @@ public ClusterOperationsImpl(OkHttpClient client, Config config, String item) {
...
-    return new VersionInfo.Builder().withBuildDate(responseAsMap.get(VersionInfo.VERSION_KEYS.BUILD_DATE))
-      .withGitCommit(responseAsMap.get(VersionInfo.VERSION_KEYS.GIT_COMMIT))
-      .withGitVersion(responseAsMap.get(VersionInfo.VERSION_KEYS.GIT_VERSION))
-      .withMajor(responseAsMap.get(VersionInfo.VERSION_KEYS.MAJOR))
-      .withMinor(responseAsMap.get(VersionInfo.VERSION_KEYS.MINOR))
-      .withGitTreeState(responseAsMap.get(VersionInfo.VERSION_KEYS.GIT_TREE_STATE))
-      .withPlatform(responseAsMap.get(VersionInfo.VERSION_KEYS.PLATFORM))
-      .withGoVersion(responseAsMap.get(VersionInfo.VERSION_KEYS.GO_VERSION))
-      .withCompiler(responseAsMap.get(VersionInfo.VERSION_KEYS.COMPILER))
+  protected VersionInfo fetchVersionInfoFromResponse(Map<String, String> responseAsMap) throws ParseException {
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ClusterOperationsImpl.java:66]: This one can be static

File: openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftClusterOperationsImpl.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    if (response.isSuccessful() && response.body() != null) {
+      ClusterVersionList clusterVersionList = objectMapper.readValue(response.body().string(), ClusterVersionList.class);
+      if (!clusterVersionList.getItems().isEmpty()) {
+        return parseVersionInfoFromClusterVersion(clusterVersionList.getItems().get(0));
+      }
+    }
+    return null;
+  }
+
+  public static VersionInfo parseVersionInfoFromClusterVersion(ClusterVersion clusterVersion) throws ParseException {
```
Comment (manusa) [inline openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftClusterOperationsImpl.java:69]: Can we make this private?

**Comment by manusa** [comment]
I can't find where the `OpenShiftReadiness` class is used

**Review by manusa — APPROVED** [review]

**Comment by manusa** [comment]
[merge]

**Pull request merged**